tests/plugins/widget/integration/selection/inputselection.js

```diff
@@ -0,0 +1,123 @@
+/* bender-tags: widgetcore */
+/* bender-ckeditor-plugins: widget,undo */
+/* bender-ckeditor-remove-plugins: tableselection */
+
+( function() {
+	'use strict';
+
+	bender.editor = {
+		config: {
+			allowedContent: true,
+			on: {
+				instanceReady: function( evt ) {
+					var editor = evt.editor;
+
+					editor.widgets.add( 'testwidget', {
+						requiredContent: 'div(testwidget)',
+						allowedContent: 'div(testwidget)',
+						template: '<div class="testwidget">' +
+							'<textarea></textarea>' +
+							'</div>',
+						button: 'Add textarea widget',
+						upcast: function( element ) {
+							return element.name === 'div' && element.hasClass( 'testwidget' );
+						},
+						init: function() {
+							this.on( 'focus', function() {
+								this.element.findOne( 'textarea' ).focus();
+							} );
+						}
+					} );
+
+					editor.widgets.add( 'testwidget2', {
```

Comment:
As the widgets are very similar, maybe it makes sense to extract them into the common builder function?